plugins/Eq/EqSpectrumView.h

```diff
@@ -202,16 +231,29 @@ class EqSpectrumView : public QWidget
 			{
 				continue;
 			}
-			if (h > m_bandHeight.at(x))
+
+			if ( h > m_bandHeight[x] )
 			{
 				m_bandHeight[x] = h;
 			}
 			else
 			{
 				m_bandHeight[x] = m_bandHeight[x] -1;
 			}
-			pp.lineTo( freqToXPixel( bandToFreq( x ) ), fh - m_bandHeight.at( x ) );
+			if( m_bandHeight[x] < 0 )
+			{
+				m_bandHeight[x] = 0;
+			}
+
+			pp.lineTo( freqToXPixel( bandToFreq( x ) ), fh - m_bandHeight[x] );
+			h_sum = h_sum + m_bandHeight[x];
 		}
+
+		if( h_sum == 0 )
+		{
+			m_sa->m_energy = 0;
```

Comment:
This code should not be in a paint event. This should be handled in `analyze()`.